src/layers/basic.jl

```diff
@@ -172,31 +172,45 @@ function Base.show(io::IO, l::Dense)
 end
 
 """
-    Diagonal(α, β)
-    Diagonal(size::Integer...)
+    Diagonal(scale, bias)
+    Diagonal(size::Integer...; bias = true, init = ones32)
 
 Create an element-wise linear layer, which performs
 
-    y = α .* x .+ β
+    y = scale .* x .+ bias
 
-The learnable arrays are initialised `α = ones(Float32, size)` and
-`β = zeros(Float32, size)`.
+if `bias` is true, and
+
+    y = x .* scale
+    
+otherwise.
+
+The learnable parameters are initialised `scale = Flux.ones32(size)` and 
+`bias = Flux.zeros32(size)`. Alternatively, specify `init` to customize how `scale` 
+is initialised. The weight matrix and/or the bias vector (with the same size as `x`) 
+may also be provided explicitly.
```

Comment:
`y = scale .* x .+ bias` is true in both cases, but it seems a little confusing to say that this is true if `bias=true`, since the `true` isn't what goes into that formula. Perhaps we can be quite brief since Dense works the same way:

```suggestion
with no activation function.
 
The learnable parameters are initialised `scale = init(size...)` and 
`bias = zeros32(size...)`, with `init = ones32` by default. You may specify the function `init`, 
turn off trainable bias with `bias = false`, or provide the array(s) explicitly.
```